Update README.rst

appetrosyan opened the pull request:
Added installation instructions from CosmoMC

appetrosyan committed:
Update README.rst

Added installation instructions from CosmoMC

williamjameshandley commented on README.rst:
```diff
@@ -55,3 +55,38 @@ The changes to CosmoMC are minor:
+  
+Planck Likelihoods
+==================
+As this is a fork of CosmoMC the process of installing the Planck lieklihoods is identical: 
```
lieklihoods -> likelihoods

williamjameshandley commented on README.rst:
```diff
@@ -55,3 +55,38 @@ The changes to CosmoMC are minor:
+Planck Likelihoods
+==================
+As this is a fork of CosmoMC the process of installing the Planck lieklihoods is identical: 
+- install prerequisites of Planck Code: 
+.. bash::
+   pip install cython astropy 
+   sudo pacman -S cfitsio
```
Here I think it's probably best to say '(you may need) to install cfitsio if the `--install_all_deps` later on is unable to', and simply have a list of prerequisites, rather than presupposing an arch-based distribution (or the use of pip for installing python packages).

appetrosyan replied:
It almost always breaks: `--install_all_deps `, since it wasn't updated. It uses hardcoded paths, and I only mention it here, so that the user knows what to pull in, and make sure it's recognised.

williamjameshandley replied:
I see. It still is better to provide advice that is python installation (i.e.
many people use conda, apt or pacman to install python)
and package manager independent. This would also presuppose root access, which
you don't have for example on a HPC machine.

williamjameshandley commented on README.rst:
```diff
@@ -55,3 +55,38 @@ The changes to CosmoMC are minor:
+   pip install cython astropy 
+   sudo pacman -S cfitsio
+   
+- Obtain from the `Planck Legacy archive <https://wiki.cosmos.esa.int/planck-legacy-archive/index.php/Main_Page>` the following the `COM_Likelihood_Code-*.tar.gz` and `COM-Likelihood_Data-*.tar.gz`.
```
Is there a direct download link? I seem to remember that cobaya uses one, so this might make the installation even more seamless for new users if they could just click a link to download it (or use curl/wget).

appetrosyan replied:
These links break often. I'd be hesitant.

williamjameshandley replied:
It seems to work for cobaya fairly well in my experience. You could always say 'use these links, but if that fails visit the PLA'

williamjameshandley commented on README.rst:
```diff
@@ -55,3 +55,38 @@ The changes to CosmoMC are minor:
+.. bash::
+   ./waf configure --install_all_deps install
+   
+- make sure that all packages are pulled in correctly. Any missing packages may not prevent the code from building, but `CosmoChord` may not have all of its dependencies. 
+- source the environment variables using a `bash`-compatible shell. e.g. 
+.. bash::
+   echo echo -e "\nsource $(pwd)/bin/clik_profile.sh" >> ~/.bashrc
```
Here I think it's probably clearer to simply say  something like 'you need to run the `source bin/clik_profile.sh` each time you want to use the Planck likelihood. If you want to avoid remembering to run this every time, you should put it into your bashrc`. I personally don't have it in my bashrc, but that's because I have had to use multiple planck likelihoods in the past, and generally tend to work in virtual environments to avoid cluttering the PATH with potentially contradictory setups.

appetrosyan replied:
It breaks on `fish` and `xonsh`. You need to use the same bash-like shell to source as the shell you use to run the commands. Ideally, we should just have `fish` and Python versions of these environment set-up scripts.

williamjameshandley replied:
Also on zsh in my experience, but if you're using those, then you're likely to be able to cope with modifying them as necessary. Surely this is an argument for providing non bash-specific advice?

williamjameshandley reviewed (changes requested):
Many thanks for this very helpful contribution @appetrosyan. I have a few minor comments/discussion points, and then we can merge this in.

williamjameshandley commented:
I see. It still is better to provide advice that is python installation (i.e.  
many people use conda, apt or pacman to install python)
and package manager independent. This would also presuppose root access, which 
you don't have for example on a HPC machine.

On Mon, Apr 20, 2020 at 07:42:09AM -0700, Aleksandr Petrosyan wrote:
>@appetrosyan commented on this pull request.
>
>
>
>> +- install prerequisites of Planck Code:
>+.. bash::
>+   pip install cython astropy
>+   sudo pacman -S cfitsio
>
>It almost always breaks. --install_all_deps wasn't updated, so it's hardcoded to download versions that are no longer supported and no longer exist.
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/williamjameshandley/CosmoChord/pull/1#discussion_r411434699

-- 
Dr Will Handley
Research Fellow at Gonville & Caius College

Cavendish Astrophysics Group
Cavendish Laboratory
JJ Thomson Avenue
Cambridge, UK
CB3 0HE

Room K16
Kavli Institute for Cosmology
Cambridge, UK

Tel: +44-(0)1223-764042
Mob: +44-(0)7718-622713
Email: wh260@cam.ac.uk
arXiv papers: https://arxiv.org/a/handley_w_1
GitHub: https://github.com/williamjameshandley
Skype: wjhandley
Website: http://www.kicc.cam.ac.uk/directory/wh260

appetrosyan committed:
Update README.rst

Implemented comments.

appetrosyan committed:
Update README.rst

Refined the wording.

appetrosyan committed:
Update README.rst

williamjameshandley commented on README.rst:
```diff
@@ -55,3 +55,58 @@ The changes to CosmoMC are minor:
+Planck Likelihoods
+==================
+As this is a fork of CosmoMC, the process of installing the Planck likelihoods is identical: 
+- install prerequisites of Planck Code: 
+
+.. bash::
+   pip install cython astropy 
+
+- install ``cfitsio``. For ArchLinux this is available in the official repositories. On Scientific linux, it's pre-installed and needs to be found with ``module avail`` and loaded with e.g. ``module load cfitsio-<somecompiler>-<someversion>``.  Make sure that the compiler version matches the one used for compiling CosmoChord, and Planck likelihood code.   
```
I think that this is still too system-specific, and new users are likely to get lost in the detail associated with arch linux (a very small subset of astrophysicists use this), scientific linux (many cluster users  won't actually know what linux distribution their cluster runs on).

To cater to the widest variety of users, it would be better simply to say something like "The prerequisites for the Planck likelihood code are that you have installed the cython and astropy Python modules, and have a working version of cfitsio (I believe if one has installed astropy, then cfitsio should also be working).

williamjameshandley commented on README.rst:
```diff
@@ -55,3 +55,58 @@ The changes to CosmoMC are minor:
+   note that if this fails, the `waf` script will attempt to pull the dependencies from obsolete hardcoded locations. 
+   If this is the case, interrupt (`Ctrl+c`) and install the dependencies manually. THis may not prevent the Placnk 
+   
+- Set-up the environment variables. An example profile for ``bash`` is given in ``bin/clik_profile.sh``. To avoid frustration, you may wish to source the profile at login, e.g.
+
+.. bash::
+   echo echo -e "\nsource $(pwd)/bin/clik_profile.sh" >> ~/.bashrc
```
It would be better to avoid a possibly confusing bash command, and simply say "put the following line in your .bashrc", which then caters to a wider variety of users and abilities.

williamjameshandley reviewed (changes requested):
This is all looking much better. There are a couple of changes that I think would make it easier for users who are newer to linux (which are generally quite strongly correlated with astrophysicists installing software for the first time, which is who a README such as this is primarily targeted at)

appetrosyan committed:
Accounted for the changes needed.

appetrosyan committed:
Fixed display of inline code

appetrosyan committed:
Fixed display of inline code and links

appetrosyan committed:
Merge branch 'master' into patch-1

appetrosyan reviewed (commented):
Resolved! Any further changes?

appetrosyan committed:
Update README.rst

Added quotes to make the `curl` script not fail on exotic shells such as `zsh` and `fish`

appetrosyan committed:
Update README.rst

Fixed path for planck code.

appetrosyan committed:
Update README.rst

Fixed Path issues.
